### PR TITLE
feat(cli): Keel data generation CLI

### PR DESCRIPTION
**Pull request by luispollo**
Introduces a simple Spring Boot console application that can be used to aid in development and testing. In this PR, a single command is defined, `datagen`, which populates the local database with test records to assist with testing.

### CODE REVIEW

File: keel-cli/src/main/kotlin/com/netflix/spinnaker/keel/cli/KeelCli.kt [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+@ComponentScan(
+  basePackages = [
+    "com.netflix.spinnaker.config",
+    "com.netflix.spinnaker.keel"
+  ],
+  excludeFilters = [ComponentScan.Filter(
+    type = FilterType.REGEX,
+    pattern = [
+      "com.netflix.spinnaker.config.DefaultServiceClientProvider",
+      "com.netflix.spinnaker.config.MetricsEndpointConfiguration",
```
Comment (luispollo) [inline keel-cli/src/main/kotlin/com/netflix/spinnaker/keel/cli/KeelCli.kt:65]: These kork components enable web-related stuff we don't want in a CLI.

File: keel-cli/src/main/kotlin/com/netflix/spinnaker/keel/cli/KeelCli.kt [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      "com.netflix.spinnaker.keel.artifacts.ArtifactListener"
```
Comment (luispollo) [inline keel-cli/src/main/kotlin/com/netflix/spinnaker/keel/cli/KeelCli.kt:66]: We don't want to try and find metadata for fake artifacts.

File: keel-cli/src/main/kotlin/com/netflix/spinnaker/keel/cli/commands/DataGenCommand.kt [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  companion object {
+    const val TEST_APP = "datagen"
+    const val TEST_STACK = "local"
+    const val TEST_ACCT = "test"
+    const val TEST_DEB_PACKAGE = "fakedeb"
+    const val TEST_SVC_ACCT = "delivery-engineering@netflix.com"
+    val TEST_ARTIFACT = DebianArtifact(
+      deliveryConfigName = TEST_APP,
+      name = TEST_DEB_PACKAGE,
+      vmOptions = VirtualMachineOptions(baseOs = "bionic", regions = setOf("us-weest-2")),
```
Comment (luispollo) [inline keel-cli/src/main/kotlin/com/netflix/spinnaker/keel/cli/commands/DataGenCommand.kt:38]:
```suggestion
      vmOptions = VirtualMachineOptions(baseOs = "bionic", regions = setOf("us-west-2")),
```

File: keel-cli/src/main/kotlin/com/netflix/spinnaker/keel/cli/commands/DataGenCommand.kt [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+      (1..artifactCount).forEach { index ->
+        println("Writing artifact version record for delivery config ${deliveryConfig.name}, environment test, version 1.0.$index")
+        repository.markAsSuccessfullyDeployedTo(deliveryConfig, TEST_ARTIFACT, "1.0.$index", "test")
+      }
+    }
+
+    println("Successfully generated test data. Happy testing!")
+
+    Thread.sleep(10 * 1000)
```
Comment (luispollo) [inline keel-cli/src/main/kotlin/com/netflix/spinnaker/keel/cli/commands/DataGenCommand.kt:131]:
```suggestion
```

**Review by lorin — APPROVED** [review]
Neat!

**Pull request merged**